src/analyze/analyze-security.c

```diff
@@ -2100,6 +2113,138 @@ static int analyze_security_one(sd_bus *bus, const char *name, Table *overview_t
         return 0;
 }
 
+/* Refactoring SecurityInfo so that it can make use of existing struct variables instead of reading from dbus */
+static int get_security_info(Unit *u, ExecContext *c, CGroupContext *g, SecurityInfo **ret_info) {
+        _cleanup_(security_info_free_freep) SecurityInfo *info = new0(SecurityInfo, 1);
+        if (!info)
+                return log_oom();
+
+        assert(ret_info);
```

Comment:
we usually check params first, and only then start doing stuff such as allocating objects. i.e. move this assert to the top of the function, please